client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -138,31 +139,43 @@ export const receiveCommentError = ( { dispatch, getState }, { siteId, commentId
 
 // @see https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/comments/
 export const fetchCommentsList = ( { dispatch }, action ) => {
-	const { query } = action;
-
-	if ( ! query || 'site' !== query.listType ) {
+	if ( ! action.query || 'site' !== action.query.listType ) {
 		return;
 	}
 
-	const { siteId, status = 'unapproved', type = 'comment' } = query;
+	const { siteId, status = 'unapproved', type = 'comment' } = action.query;
+
+	const query = {
+		...omit( action.query, [ 'listType', 'siteId' ] ),
+		status,
+		type,
+	};
 
 	dispatch(
 		http(
 			{
 				method: 'GET',
 				path: `/sites/${ siteId }/comments`,
 				apiVersion: '1.1',
-				query: {
-					status,
-					type,
-				},
+				query,
 			},
 			action,
 		),
 	);
 };
 
-export const addComments = ( { dispatch }, { query: { siteId } }, next, { comments } ) => {
+export const addComments = ( { dispatch }, { query: { siteId, status } }, next, { comments } ) => {
+	// Initialize the comments tree
```

Comment:
Might actually be more accurate to say something like "Make sure comments tree is initialized"

actually, why do we need to do this if we are also listening for these actions and storing a separate `isInitialized` state value?